src/statespace/SO2.cpp

```diff
@@ -3,37 +3,38 @@
 namespace aikido {
 namespace statespace {
 //==============================================================================
-SO2::State::State() : mAngle(0.)
+SO2::State::State(double angle)
 {
+  fromAngle(angle);
 }
 
 //==============================================================================
-SO2::State::State(double _angle) : mAngle(_angle)
-{
-}
-
-//==============================================================================
-double SO2::State::getAngle() const
+double SO2::State::toAngle() const
 {
   return mAngle;
 }
 
 //==============================================================================
-void SO2::State::setAngle(double _angle)
+void SO2::State::fromAngle(double angle)
 {
-  mAngle = _angle;
+  double boundedAngle = std::fmod(angle, 2.0 * M_PI);
+  if (boundedAngle > M_PI)
+    boundedAngle -= 2.0 * M_PI;
+  if (boundedAngle < -M_PI)
```

Comment:
It looks like this permits `boundedAngle` to be `-M_PI`, while the documentation says that it is bounded in `(-pi, pi]`.